pkg/specgen/generate/kube/volume.go

```diff
@@ -98,23 +107,64 @@ func VolumeFromPersistentVolumeClaim(claim *v1.PersistentVolumeClaimVolumeSource
 	}, nil
 }
 
+func VolumeFromConfigMap(configMapVolumeSource *v1.ConfigMapVolumeSource, configMaps []v1.ConfigMap) (*KubeVolume, error) {
+	var configMap *v1.ConfigMap
+	kv := &KubeVolume{Type: KubeVolumeTypeConfigMap, Items: map[string]string{}}
+	for _, cm := range configMaps {
+		if cm.Name == configMapVolumeSource.Name {
+			configMap = &cm
+			// Set the source to the config map name
+			kv.Source = cm.Name
+			break
+		}
+	}
+
+	if configMap == nil {
+		// If the volumeSource was optional, move on even if a matching configmap wasn't found
+		if *configMapVolumeSource.Optional {
+			kv.Source = configMapVolumeSource.Name
+			kv.Optional = *configMapVolumeSource.Optional
+			return kv, nil
+		} else {
+			return nil, errors.Errorf("no such ConfigMap %q", configMapVolumeSource.Name)
+		}
+	}
+
+	// If there are Items specified in the volumeSource, that overwrites the Data from the configmap
+	if len(configMapVolumeSource.Items) > 0 {
+		for _, item := range configMapVolumeSource.Items {
+			if val, ok := configMap.Data[item.Key]; ok {
```

Comment:
Won't this config.Data blow up if the configMap == nil above happens?

Reply:
Nope, we always return before getting to this point if configMap is nil.